app/models/subject.rb

```diff
@@ -1,5 +1,7 @@
 class Subject < ActiveRecord::Base
   has_and_belongs_to_many :posts
+  validates :name, presence: true, uniqueness: true
```

Comment:
I see you have the uniqueness here but could you also create a migration to add it to the db?